main/sdr.go

```diff
@@ -319,20 +312,96 @@ func sdrKill() {
 	}
 }
 
-// Watch for config/device changes.
-func sdrWatcher() {
-	var doSkip bool
-	rES, err := regexp.Compile("str?a?t?u?x:1090")
-	if err != nil {
-		rES = nil
-		log.Println("failed to compile ES regexp because %s", err.Error())
+func reCompile(s string) *regexp.Regexp {
+	// note , compile returns a nil pointer on error
+	r, _ := regexp.Compile(s)
+	return r
+}
+
+type regexUAT regexp.Regexp
+type regexES regexp.Regexp
+
+var rUAT = (*regexUAT)(reCompile("str?a?t?u?x:978"))
+var rES = (*regexUAT)(reCompile("str?a?t?u?x:1090"))
```

Comment:
On Mon, Feb 22, 2016 at 7:39 AM, peepsnet <notifications@github.com> wrote:

> In main/sdr.go
> <https://github.com/cyoung/stratux/pull/279#discussion_r53624975>:
>
> > +type regexUAT regexp.Regexp
> > +type regexES regexp.Regexp
> > +
> > +var rUAT = (*regexUAT)(reCompile("str?a?t?u?x:978"))
> > +var rES = (*regexUAT)(reCompile("str?a?t?u?x:1090"))
>
> I love the copy and paste errors!! They always kill me!
>
I'm usually pretty good about following a fairly strict workflow to avoid
this sort of stuff... :(